### PR TITLE
AIP-7314 Argo Flow Trigger Flow

### PR DESCRIPTION
**Pull request by talebzeghmi**

### CODE REVIEW

File: metaflow/plugins/kfp/kfp.py [hunk 1]
```diff
@@ -70,7 +69,7 @@ class FlowVariables:
     environment: str
     event_logger: str
     monitor: str
-    namespace: str
+    user_namespace: str
```
Comment (cloudw) [inline metaflow/plugins/kfp/kfp.py:72]: Are you trying to distinguish `metaflow_namespace` v.s. `k8s_namespace` here?

File: metaflow/plugins/kfp/kfp_cli.py [hunk 2]
```diff
@@ -327,6 +334,314 @@ def _convert_value(param: parameters.Parameter):
...
+)
+@click.option(
+    "--notify-on-success",
+    "-nos",
+    "notify_on_success",
+    default=from_conf("METAFLOW_NOTIFY_ON_SUCCESS", default=None),
+    help="Email address to notify upon success"
+    "If not set, METAFLOW_NOTIFY_ON_SUCCESS is used from Metaflow config or environment variable",
+    show_default=True,
+)
```
Comment (cloudw) [inline metaflow/plugins/kfp/kfp_cli.py]: It looks like a lot of these options are duplicated from the `run` command. Can we reduce the duplications here? This SO question may be helpful: https://stackoverflow.com/questions/50061342/is-it-possible-to-reuse-python-click-option-decorators-for-multiple-commands

File: metaflow/plugins/kfp/kfp_cli.py [hunk 3]
```diff
@@ -327,6 +334,314 @@ def _convert_value(param: parameters.Parameter):
...
+    )
+    argo_run_id = workflow["metadata"]["name"]
+    metaflow_run_id = to_metaflow_run_id(argo_run_id)
+
+    if run_id_file:
+        with open(run_id_file, "w") as f:
+            f.write(str(argo_run_id))
+
+    obj.echo(
+        f"Workflow *{workflow_name}* triggered on Argo Workflows "
```
Comment (cloudw) [inline metaflow/plugins/kfp/kfp_cli.py]: Do we expect `argo_workflow_name` to differ from `workflow_name` in this function? Also `argo_run_id` seems to be the same as `argo_workflow_name`

File: metaflow/plugins/kfp/argo_client.py [hunk 4]
```diff
@@ -196,3 +222,18 @@ def run_workflow(self, workflow: Dict[str, Any]) -> Dict[str, Any]:
             raise ArgoClientException(
                 json.loads(e.body)["message"] if e.body is not None else e.reason
             )
+
+    def delete_workflow(self, name: str):
```
Comment (cloudw) [inline metaflow/plugins/kfp/argo_client.py]: This should be named as `delete_workflow_template`.

**Review by cloudw — CHANGES_REQUESTED** [review]

File: metaflow/plugins/kfp/argo_client.py (hunk 4 above)
Comment (talebzeghmi) [reply metaflow/plugins/kfp/argo_client.py]: done!

File: metaflow/plugins/kfp/kfp.py (hunk 1 above)
Comment (talebzeghmi) [reply metaflow/plugins/kfp/kfp.py:72]: yes, very much so, the user selected metaflow namespace to run in.

File: metaflow/plugins/kfp/kfp_cli.py (hunk 3 above)
Comment (talebzeghmi) [reply metaflow/plugins/kfp/kfp_cli.py]: removed workflow_name and argo_run_id replacing with argo_workflow_name to reduce confusion

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -9,6 +9,7 @@ on:
     - master
     - feature/kfp
     - feature/kfp-argo
+    - tz/AIP-7299-argo-submit  # TODO(talebz) remove before merging
```
Comment (cloudw) [inline .github/workflows/test.yml:12]: Reminder

File: metaflow/plugins/kfp/argo_utils.py [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+
+def run_argo_workflow(
+    kubernetes_namespace: str,
+    template_name: str,
+    parameters: Optional[dict] = None,
+    wait_timeout: Union[int, float, datetime.timedelta] = 0,
+    **kwarg,  # Other parameters for wait function
+) -> str:
+    try:
+        # TODO(talebz): add tag of origin-run-id to correlate parent flow
```
Comment (cloudw) [inline metaflow/plugins/kfp/argo_utils.py:23]: Do you intend to resolve this before PR merging, or do you have a ticket created?

File: metaflow/plugins/kfp/kfp.py (hunk 1 above)
Comment (cloudw) [reply metaflow/plugins/kfp/kfp.py:72]: I'm feeling the `user_namespace` still isn't explicit enough. You used to have `namespace` referring to metalfow namespace to align with the metaflow open source code, and `k8s_namespace` means k8s ones specifically. Though it looks like the open sources code does use `namespace` for k8s ones (for example https://github.com/Netflix/metaflow/blob/master/metaflow/plugins/argo/argo_client.py).

Not for this PR, but can we maybe push for some var renaming all the way into the open sourced code to reduce the confusion?

I just want to make sure this is a deliberated renaming. I can align my mental dictionary with yours if you feel the renaming is necessary.

File: metaflow/plugins/kfp/kfp.py [hunk 7]
```diff
@@ -222,7 +271,7 @@ def _create_workflow_yaml(
             # self.name is typically CamelCase as it's python class name.
             # generateName contains a sanitized version of self.name from kfp.compiler
             workflow["metadata"]["name"] = (
-                workflow["metadata"].pop("generateName").rstrip("-")
+                name if name else workflow["metadata"].pop("generateName").rstrip("-")
```
Comment (cloudw) [inline metaflow/plugins/kfp/kfp.py:274]: nit: Might be less readable but can use `or`: `name or workflow["metadata"].pop("generateName").rstrip("-")`

**Review by cloudw — APPROVED** [review]
A few more comments but LGTM.

File: metaflow/plugins/kfp/argo_utils.py (hunk 6 above)
Comment (talebzeghmi) [reply metaflow/plugins/kfp/argo_utils.py:23]: created https://zbrt.atl.zillow.net/browse/AIP-7382

**Comment by talebzeghmi** [comment]
> I'm feeling the user_namespace still isn't explicit enough. You used to have namespace referring to metalfow namespace to align with the metaflow open source code, and k8s_namespace means k8s ones specifically. Though it looks like the open sources code does use namespace for k8s ones (for example https://github.com/Netflix/metaflow/blob/master/metaflow/plugins/argo/argo_client.py).
> 
> Not for this PR, but can we maybe push for some var renaming all the way into the open sourced code to reduce the confusion?
> 
> I just want to make sure this is a deliberated renaming. I can align my mental dictionary with yours if you feel the renaming is necessary.

good points!
see https://zbrt.atl.zillow.net/browse/AIP-7383 for

**Pull request merged**